File: lollys-lotto-ts-sdk/src/codegen/instructions/createEventEmitter.ts
import { TransactionInstruction, PublicKey, AccountMeta } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@coral-xyz/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as types from "../types" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CreateEventEmitterAccounts {
  /** Lamports for rent funded from here. */
  funder: PublicKey
  eventEmitter: PublicKey
  /** Needed to create a new account */
  systemProgram: PublicKey
}

export function createEventEmitter(
  accounts: CreateEventEmitterAccounts,
  programId: PublicKey = PROGRAM_ID
) {
  const keys: Array<AccountMeta> = [
    { pubkey: accounts.funder, isSigner: true, isWritable: true },
    { pubkey: accounts.eventEmitter, isSigner: false, isWritable: true },
    { pubkey: accounts.systemProgram, isSigner: false, isWritable: false },
  ]
  const identifier = Buffer.from([177, 70, 19, 205, 180, 108, 160, 187])
  const data = identifier
  const ix = new TransactionInstruction({ keys, programId, data })
  return ix
}

File: lollys-lotto-rust-sdk/src/instructions/admin/create_event_emitter.rs
use crate::instructions::*;

pub fn create_event_emitter(event_emitter: &Pubkey, authority: &Pubkey) -> Instruction {
    let data = lollys_lotto::instruction::CreateEventEmitter.data();
    let accounts = lollys_lotto::accounts::CreateEventEmitter {
        funder: *authority,
        event_emitter: *event_emitter,
        system_program: system_program::ID,
    }
    .to_account_metas(None);
    Instruction {
        program_id: lollys_lotto::id(),
        accounts,
        data,
    }
}
